dev/integration_tests/ios_add2app_life_cycle/ios_add2app/MainViewController.m

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "MainViewController.h"
+
+#import "AppDelegate.h"
+#import "FullScreenViewController.h"
+
+@interface MainViewController ()
+
+@end
+
+
+@implementation MainViewController {
+  UIStackView *_stackView;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  [self.view setFrame:self.view.window.bounds];
+  self.title = @"Flutter iOS Demos";
+  self.view.backgroundColor = UIColor.whiteColor;
+
+  _stackView = [[UIStackView alloc] initWithFrame:self.view.frame];
+  _stackView.axis = UILayoutConstraintAxisVertical;
+  _stackView.distribution = UIStackViewDistributionEqualSpacing;
+  _stackView.alignment = UIStackViewAlignmentCenter;
+  _stackView.autoresizingMask =
+      UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+  _stackView.layoutMargins = UIEdgeInsetsMake(0, 0, 50, 0);
+  _stackView.layoutMarginsRelativeArrangement = YES;
+  [self.view addSubview:_stackView];
+
+  [self addButton:@"Full Screen (Cold)" action:@selector(showFullScreenCold)];
+}
+
+- (FlutterEngine *)engine {
+  return [(AppDelegate *)[[UIApplication sharedApplication] delegate] engine];
+}
+
+- (FlutterBasicMessageChannel*)reloadMessageChannel {
```

Comment:
Sorry I'm not familiar with the message channel patterns, is this used anywhere?